spec/datadog/tracing/client_ip_spec.rb

```diff
@@ -0,0 +1,344 @@
+require 'spec_helper'
+
+require 'datadog/core/header_collection'
+require 'datadog/tracing/client_ip'
+require 'datadog/tracing/metadata/ext'
+
+RSpec::Matchers.define :be_valid_ip do
+  match do |actual|
+    normalised = Datadog::Tracing::ClientIp.strip_decorations(actual)
+    begin
+      Datadog::Tracing::ClientIp.validate_ip(normalised)
+
+      true
+    rescue Datadog::Tracing::ClientIp::InvalidIpError
+      false
+    end
+  end
+end
+
+RSpec.describe Datadog::Tracing::ClientIp do
+  subject(:client_ip) { described_class }
+  let(:tagging_enabled) { true }
+  let(:ip_header_name) { nil }
+
+  before do
+    Datadog.configure do |c|
+      c.tracing.client_ip.disabled = !tagging_enabled
+      c.tracing.client_ip.header_name = ip_header_name
+    end
+  end
+
+  after do
+    without_warnings { Datadog.configuration.reset! }
+  end
+
+  describe 'ip validation' do
+    context 'when given valid ip addresses' do
+      subject do
```

Comment:
I think this could be simplified, without custom matcher
```
describe ".validate_ip" do
  ips = [
    ...
  ]

  ips.each do |ip|
    context 'when given #{ip}' do
      let(:normalised) { described_class.strip_decorations(ip) }
      subject { described_class.validate_ip(normalised) }

      it { is_expected.to be true }
    end
  end
end
```

Reply:
Fixed